src/tla2528.cpp

```diff
@@ -0,0 +1,186 @@
+#include <bitset>
+#include <libhal-util/bit.hpp>
+#include <libhal-util/i2c.hpp>
+#include <libhal/error.hpp>
+#include <libhal/units.hpp>
+#include <libhal-expander/tla2528.hpp>
+
+namespace {
+enum op_codes : hal::byte
+{
+  single_register_read = 0b0001'0000,
+  single_register_write = 0b0000'1000,
+  set_bit = 0b0001'1000,
+  clear_bit = 0b0010'0000,
+  // Continuously reads data from a group of registers. Provide the first
+  // address to read from, if it runs out of valid addresses to read, it
+  // returns zeros. (See Figure 30 on datasheet)
+  continuous_register_read = 0b0011'0000,
+  // Continuously writes data to a group of registers. Provide the first
+  // address to write to.The data sent will automatically write the data to
+  // the next register in ascending order. (See Figure 32 on datasheet)
+  continuous_register_write = 0b0010'1000
+};
+
+enum register_addresses : hal::byte
+{
+  system_status = 0x0,
+  general_cfg = 0x1,
+  data_cfg = 0x2,
+  osr_cfg = 0x3,
+  opmode_cfg = 0x4,
+  pin_cfg = 0x5,
+  gpio_cfg = 0x7,
+  gpo_drive_cfg = 0x9,
+  gpo_value = 0xB,
+  gpi_value = 0xD,
+  sequence_cfg = 0x10,
+  channel_sel = 0x11,
+  auto_seq_ch_sel = 0x12
+};
+}  // namespace
+namespace hal::expander {
+
+tla2528::tla2528(hal::i2c& p_i2c, hal::byte p_i2c_address)
+  : m_i2c_bus(p_i2c)
+  , m_i2c_address(p_i2c_address)
+{
+  // TODO: reset command
```

Comment:
```suggestion
  // TODO(#issue-number): reset command
```

Make an technical debt issue for this. Also you might as well call reset() here and define it with the TODO. That way the code is already in place and the empty function has the todo embedded in it.